src/core/trtserver.cc

```diff
@@ -389,24 +392,29 @@ class TrtServerRequestOptions {
   TRTSERVER_Error* AddOutput(const char* output_name);
   TRTSERVER_Error* AddOutput(const char* output_name, uint32_t count);
 
-  TRTSERVER_Error* Normalize(ni::InferenceBackend* backend);
-
   const std::string& ModelName() const { return model_name_; }
   int64_t ModelVersion() const { return model_version_; }
   ni::InferRequestHeader* InferRequestHeader() const;
+#ifdef TRTIS_ENABLE_GRPC_V2
+  const std::string& IdStr() const { return id_str_; }
+#endif  // TRTIS_ENABLE_GRPC_V2
 
  private:
   const std::string model_name_;
   const int64_t model_version_;
+
+#ifdef TRTIS_ENABLE_GRPC_V2
+  std::string id_str_;
+#endif  // TRTIS_ENABLE_GRPC_V2
+
   std::shared_ptr<ni::InferRequestHeader> request_header_;
```

Comment:
Also replace this with `InferenceRequest`? Then it becomes the server front-end's responsibility to set the options according to the `InferRequestHeader` string it received.

Reply:
Yes, also part of future changes. Want to replace the entire options and provider objects with a single TRTSERVER_InferenceRequest object that is used for both request and response and that can be mutated and used across multiple requests.